packages/fxa-auth-server/lib/routes/emails.js

```diff
@@ -586,108 +558,92 @@ module.exports = (
+            code: emailData.emailCode,
+            deviceId: sessionToken.deviceId,
+            acceptLanguage: request.app.acceptLanguage,
+            email: emailData.email,
+            primaryEmail,
+            ip,
+            location: geoData.location,
+            timeZone: geoData.timeZone,
+            uaBrowser: sessionToken.uaBrowser,
+            uaBrowserVersion: sessionToken.uaBrowserVersion,
+            uaOS: sessionToken.uaOS,
+            uaOSVersion: sessionToken.uaOSVersion,
+            uid,
           });
+        } catch (err) {
+          log.error('mailer.sendVerifySecondaryEmail', { err: err });
+          await db.deleteEmail(emailData.uid, emailData.normalizedEmail);
+          throw emailUtils.sendError(err, true);
         }
 
-        function createEmail() {
-          return db.createEmail(uid, emailData);
-        }
+        return {};
 
-        function sendEmailVerification() {
-          const geoData = request.app.geo;
-          return mailer
-            .sendVerifySecondaryEmail([emailData], sessionToken, {
-              code: emailData.emailCode,
-              deviceId: sessionToken.deviceId,
-              acceptLanguage: request.app.acceptLanguage,
-              email: emailData.email,
-              primaryEmail,
-              ip,
-              location: geoData.location,
-              timeZone: geoData.timeZone,
-              uaBrowser: sessionToken.uaBrowser,
-              uaBrowserVersion: sessionToken.uaBrowserVersion,
-              uaOS: sessionToken.uaOS,
-              uaOSVersion: sessionToken.uaOSVersion,
-              uid,
-            })
-            .catch(err => {
-              log.error('mailer.sendVerifySecondaryEmail', { err: err });
-              return db
-                .deleteEmail(emailData.uid, emailData.normalizedEmail)
-                .then(() => {
-                  throw emailUtils.sendError(err, true);
+        async function deleteAccountIfUnverified() {
+          try {
+            const secondaryEmailRecord = await db.getSecondaryEmail(email);
+            if (secondaryEmailRecord.isPrimary) {
+              if (secondaryEmailRecord.isVerified) {
+                throw error.verifiedPrimaryEmailAlreadyExists();
+              }
+
+              const msSinceCreated =
+                Date.now() - secondaryEmailRecord.createdAt;
+              const minUnverifiedAccountTime =
+                config.secondaryEmail.minUnverifiedAccountTime;
+              if (msSinceCreated >= minUnverifiedAccountTime) {
+                await db.deleteAccount(secondaryEmailRecord);
+                log.info('accountDeleted.unverifiedSecondaryEmail', {
+                  ...secondaryEmailRecord,
                 });
-            });
+                return;
+              } else {
+                throw error.unverifiedPrimaryEmailNewlyCreated();
+              }
+            }
+
+            // Only delete secondary email if it is unverified and does not belong
```

Comment:
Does this mean we remove a secondary email address from another account if it's not verified?

Reply:
Whoa, it certainly looks that way.  😟

Reply:
(Note that this is in prod: https://github.com/mozilla/fxa/blob/master/packages/fxa-auth-server/lib/routes/emails.js#L636)

Reply:
There's a test for it: https://github.com/mozilla/fxa/blob/master/packages/fxa-auth-server/test/remote/recovery_email_emails.js#L91-L144.  So I suppose that's correct then!